SourceX/controls/plrctrls.cpp

```diff
@@ -203,6 +213,75 @@ void CheckMonstersNearby()
 	}
 }
 
+void FindMeleeTarget()
+{
+	bool visited[MAXDUNX][MAXDUNY] = { 0 };
+	int maxSteps = 25;
+	int rotations = 5;
+	int x = plr[myplr]._px;
+	int y = plr[myplr]._py;
+
+	std::list<bfsNode> queue;
+
+	visited[x][y] = true;
+	queue.push_back({ x, y, 0 });
+
+	while (!queue.empty()) {
+		bfsNode node = queue.front();
+		queue.pop_front();
+
+		for (int i = 0; i < 8; i++) {
+			int dx = node.x + pathxdir[i];
+			int dy = node.y + pathydir[i];
+
+			if (visited[dx][dy])
+				continue; // already visisted
+

```

Comment:
Perhaps `visited[dx][dy] = true;` should be done here?
As it stands, there are some conditions where we de-queue a node but don't set `visited[dx][dy] = true`